src/groupeddataframe/grouping.jl

```diff
@@ -87,7 +90,7 @@ function groupby(df::AbstractDataFrame, cols::Vector;
         permute!(df_groups.starts, group_perm)
         Base.permute!!(df_groups.stops, group_perm)
     end
-    GroupedDataFrame(df, cols, df_groups.rperm,
+    GroupedDataFrame(df, index(df)[cols], df_groups.rperm,
```

Comment:
We could write `intcols = index(df)[cols]` at the start of this method. Then `sdf = df[intcols]` will be faster and we do not need the second `groupby` method below` and can omit `::Vector` qualifier in this method signature for `cols`.

In short I would pass checking validity of `cols` to `index(df)[cols]` call and later work on what it returns. In this way we have a single point where column checking is done (in `getindex` for `index(df)`).

Reply:
Good idea, but the second method is needed AFAICT, otherwise we extract a single vector instead of a data frame.

Reply:
Right.